api/lib/infrastructure/repositories/certification-course-repository.js

```diff
@@ -37,6 +37,21 @@ module.exports = {
     }
   },
 
+  async getCreationDate(id) {
+    try {
+      const createdAt = await CertificationCourseBookshelf
+        .where({ id })
+        .fetch({ columns: ['createdAt'] })
+        .then((c) => c.attributes.createdAt);
+      return createdAt;
+    } catch (bookshelfError) {
+      if (bookshelfError instanceof CertificationCourseBookshelf.NotFoundError) {
+        throw new NotFoundError(`Certification course of id ${id} does not exist.`);
+      }
+      throw bookshelfError;
```

Comment:
2 choses :
1) J'ai l'impression qu'on rentre difficilement dans le catch ici car il manque l'option `{ require: true }` à `.fetch()` (donc renvoyer 0 résultat c'est ok pour lui). J'ai même l'impression que si tu pars en exception, c'est peut-être parce que `c.attributes` est `undefined`...
2) Maintenant que je suis bien  au fait de l'overhead mémoire de l'usage de bookshelf, je propose d'utiliser knex, surtout qu'ici on ne récupère qu'un seul attribut 
```js
const rows = await knex('certification-courses')
              .select('createdAt')
              .where({ id })
              .limit(1);
if(rows.length === 0) {
  throw new NotFoundError(`Certification course of id ${id} does not exist.`);
}

return rows[0].createdAt;
```

Reply:
OK ca me va bien, on peut meme utiliser first()
(du coup jai rajouté un test pour gerer l'exception)

    const row = await knex('certification-courses')
      .select('createdAt')
      .where({ id })
      .first();
    if (!row) {
      throw new NotFoundError(`Certification course of id ${id} does not exist.`);
    }

    return row.createdAt;
``
``